src/gdb_main.c

```diff
@@ -128,12 +129,17 @@ target_controller_s gdb_controller = {
 	.gettimeofday = hostio_gettimeofday,
 	.isatty = hostio_isatty,
 	.system = hostio_system,
+#endif
 };
 
 /* execute gdb remote command stored in 'pbuf'. returns immediately, no busy waiting. */
 
 int gdb_main_loop(target_controller_s *tc, char *pbuf, size_t pbuf_size, size_t size, bool in_syscall)
 {
+#if PC_HOSTED == 1
```

Comment:
Are these actually unused under BMDA? We're surprised it they are that it had been building without errors.

Reply:
Yes, once I hid the `hostio_reply()` reference on F-packet case, gcc-12 threw a couple -Wunused
```
gdb_main.c: In function ‘gdb_main_loop’:
gdb_main.c:137:40: error: unused parameter ‘tc’ [-Werror=unused-parameter]
  137 | int gdb_main_loop(target_controller_s *tc, char *pbuf, size_t pbuf_size, size_t size, bool in_syscall)
      |                   ~~~~~~~~~~~~~~~~~~~~~^~
gdb_main.c:137:92: error: unused parameter ‘in_syscall’ [-Werror=unused-parameter]
  137 | int gdb_main_loop(target_controller_s *tc, char *pbuf, size_t pbuf_size, size_t size, bool in_syscall)
      |                                                                                            ^
```
Of course I built BMDA and ran a semihosting testbench sketch against it. It does not need gdb transport hostio, and uses direct host syscalls instead. I believe it was dead code, but I don't have code coverage reports. It can be measured with gcov if need be.

Reply:
Ah, perfect. That works for us.

Reply:
ok. BMF needs gdb_hostio to e.g. read a file on the host. BMDA can read files itself, as it's running on the host. So no hostio fields needed.